src/useImageDimensions.ts

```diff
@@ -0,0 +1,61 @@
+import {useEffect, useState} from 'react'
+import {Image, ImageRequireSource} from 'react-native'
+
+export interface URISource {
+  uri: string
+}
+
+/**
+ * @param source either a remote URL or a local file resource.
+ * @returns original image dimensions (width, height and aspect ratio).
+ */
+function useImageDimensions(source: ImageRequireSource | URISource) {
+  const [[dimensions, error], setState] = useState<
+    [{width: number; height: number}?, Error?]
+  >([])
+
+  useEffect(() => {
+    try {
+      if (typeof source === 'number') {
+        const {width, height} = Image.resolveAssetSource(source)
+        setState([{width, height}])
+      } else if (typeof source === 'object' && source.uri) {
+        setState([])
+        Image.getSize(
+          source.uri,
+          (width, height) => setState([{width, height}]),
+          e => setState([dimensions, e]),
+        )
+      } else {
+        throw new Error('not implemented')
+      }
+    } catch (e) {
+      setState([dimensions, e])
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [source])
+
+  return {
+    dimensions:
+      dimensions &&
+      (Object.setPrototypeOf(dimensions, {
```

Comment:
What is the reson for using `Object.setPrototypeOf` here?

Reply:
This is to add the field 'aspectRatio' without copying the whole object.

Reply:
The true reason is to not disturb the garbage collector once again.

Reply:
> The true reason is to not disturb the garbage collector once again.

But you are running this on every render, and you are creating a new parent object every time 🤔 

It seems like it would be easier both for a human to read, and for a compiler to optimise if you just set `aspectRatio` directly in your state. That way no extra code have to run on each render loop

Reply:
Ok, fixed